src/code/z_elf_message.c

```diff
@@ -1,60 +1,68 @@
 #include "global.h"
-#include "z64elf_message.h"
-
-ElfMessage sChildSariaMsgs[] = {
-    ELF_MSG_STRENGTH_UPG(SKIP, 3, false, 0),
-    ELF_MSG_FLAG(CHECK, 0x61, false, EVENTCHKINF_37),
-    ELF_MSG_END(0x64),
-    ELF_MSG_FLAG(CHECK, 0x62, false, EVENTCHKINF_25),
-    ELF_MSG_FLAG(CHECK, 0x63, false, EVENTCHKINF_37),
-    ELF_MSG_FLAG(CHECK, 0x65, false, EVENTCHKINF_43),
-    ELF_MSG_MEDALLION(CHECK, 0x66, false, ITEM_MEDALLION_FOREST),
-    ELF_MSG_MEDALLION(CHECK, 0x66, false, ITEM_MEDALLION_FIRE),
-    ELF_MSG_MEDALLION(CHECK, 0x66, false, ITEM_MEDALLION_WATER),
-    ELF_MSG_SONG(CHECK, 0x67, false, ITEM_SONG_STORMS),
-    ELF_MSG_MEDALLION(CHECK, 0x68, false, ITEM_MEDALLION_SPIRIT),
-    ELF_MSG_MEDALLION(CHECK, 0x68, false, ITEM_MEDALLION_SHADOW),
-    ELF_MSG_END(0x69),
+#include "z64hint_commands.h"
+
+HintCmd sChildSariaHints[] = {
+    HINT_CMD_STRENGTH_UPG(HINT_CMD_TYPE_SKIP, 0, false, 3),
+    HINT_CMD_FLAG(HINT_CMD_TYPE_CHECK, EVENTCHKINF_37, false, 0x61),
```

Comment:
Especially since it's literally 2 files that use them.

Reply:
beside searchability, imo the concat just makes things harder to understand. When diving into this system for the first time yesterday and tried to learn how it worked, I had to keep asking myself "where the hell is this coming from" and reverse engineer the macro constructs to find out how it works. imo its unnecessary

I dont agree that its hurting read and writeability, its a few extra characters on the type and thats it

Reply:
If you want to understand how the macros work then yeah it adds a small layer of complexity, but that's so that using the macros is easier. It's also not especially difficult to see where these come from by inspecting the macro definitions, for the user that has great need to know how it's structured internally. Simplifying how the macros appear in the arrays should be of utmost priority, considering almost all macros use this field.

I wouldn't claim it hurts readability if it doesn't actually impact my reading of these scripts in practice. I think it's far messier than it was before, at any rate.
14 extra characters per line when each line is on average 72 characters is quite a lot (about 20%)

Reply:
to use the system youre gonna need to understand how each part works anyway. when you see CHECK everywhere in the commands and go to search that on its own, its not gonna take you to the information you need. match case floods you with bgcheck defines, match whole word doesnt bring you there cause the definition is not a whole word. its just a mess

Reply:
Searching isn't the right tool for the job in the first place, if you use go to definition and see how the macro uses it you get there right away

Reply:
you cant go to the definition of CHECK anyway cause intellisense isnt going to resolve what its concatenated from. I want to be able to directly go and see what check means without jumping through hoops

idk we will probably just need more opinions to tie breaker this, cause we just see this very differently

Reply:
given that `HINT_CMD_TYPE_SKIP` is only used once I wonder if type should be incorporated into the macro name instead of being an argument.

Reply:
this isnt only related to skip, its for all the command types
I wanted to make the command type be the actual macro as opposed to an arg, but the thing that is being checked against (flag/item/song, whatever) determines what the command structure is (as in, what combination of B0/B1/B packing arrangement it has)

So, its not really viable at least not in a clean way. At least as far as I know

Reply:
I'm generally against concatenating enum names with macros, but in cases like this I think it's fine because you would never need to search/find the full enum name unless you're looking at the internal system implementation, at which point you would need to look at and understand the macros anyway. It's very different from something like the item enum names which you need to use both in the macro and in other places in the code when using the system "externally"

Reply:
as a compromise I can bring it back for the cmd type which does matter for the scripts, but I dont feel its necessary to concat the enum names used internally in the macro that you cant see from the script anyway

Reply:
Would it be more bearable if `HINT_CMD_TYPE_` was shortened

otherwise I'm okay with fig's compromise